Reverting build strategy

Swat-SomeBug opened the pull request:
- Build only liba{benchmark_main}
  - liba{benchmark_main} exports lib{benchmark}
  - Removing disable of CRT_SECURE_WARNINGS
  - exe{link_main_test} links with liba{benchmark_main}
Updating README

Fixes #11

Swat-SomeBug committed:
Reverting build strategy
  - Build only liba{benchmark_main}
  - liba{benchmark_main} exports lib{benchmark}
  - Removing disable of CRT_SECURE_WARNINGS
  - exe{link_main_test} links with liba{benchmark_main}
Updating README

boris-kolpackov commented on src/buildfile:
```diff
@@ -2,23 +2,22 @@ intf_libs = # Interface dependencies.
+#   windows_additional_defines = '-D_CRT_SECURE_NO_WARNINGS'
 
-cxx.poptions += $windows_additional_defines
+# cxx.poptions += $windows_additional_defines
```
There is nothing wrong with disabling this while building the library itself so I would uncomment this if it helps.

boris-kolpackov commented on src/buildfile:
```diff
@@ -37,17 +36,21 @@ cxx.libs = $intf_libs $extra_libs
+lib{benchmark}:
 {
-  cxx.export.poptions += "-I$out_root/include" "-I$src_root/include" $windows_additional_defines
+  # cxx.export.poptions += "-I$out_root/include" "-I$src_root/include" $windows_additional_defines
```
I would drop this completely not to tempt anyone to uncomment it. This is definitely wrong on the conceptual level.

Swat-SomeBug replied:
Done

boris-kolpackov commented on src/buildfile:
```diff
@@ -61,10 +64,10 @@ hxx{*}: install = false
 {
   lib{benchmark}: bin.lib.version = @"-$version.project_id"
-  lib{benchmark_main}: bin.lib.version = @"-$version.project_id"
+  liba{benchmark_main}: bin.lib.version = @"-$version.project_id"
```
Library version is meaningless for a static library so I would drop it.

Swat-SomeBug replied:
Done

boris-kolpackov commented on src/buildfile:
```diff
@@ -61,10 +64,10 @@ hxx{*}: install = false
 {
   lib{benchmark}: bin.lib.version = @"-$version.major.$version.minor"
-  lib{benchmark_main}: bin.lib.version = @"-$version.major.$version.minor"
+  liba{benchmark_main}: bin.lib.version = @"-$version.major.$version.minor"
```
Ditto.

boris-kolpackov reviewed (approved):
Otherwise LGTM.

Swat-SomeBug committed:
Updated according to review
  - Removed $windows_additional_defines -> not exported anymore
  - Removed versioning for static lib

Swat-SomeBug committed:
Release version 1.7.0+1
